mongo/integration/handshake_test.go

```diff
@@ -199,3 +200,53 @@ func TestHandshakeProse(t *testing.T) {
 		})
 	}
 }
+
+func TestLoadBalancedConnectionHandshake(t *testing.T) {
+	mt := mtest.New(t)
+
+	lbopts := mtest.NewOptions().ClientType(mtest.Proxy).Topologies(
+		mtest.LoadBalanced)
+
+	mt.RunOpts("LB connection handshake uses OP_MSG", lbopts, func(mt *mtest.T) {
+		// Ping the server to ensure the handshake has completed.
+		err := mt.Client.Ping(context.Background(), nil)
+		require.NoError(mt, err, "Ping error: %v", err)
+
+		messages := mt.GetProxiedMessages()
+		handshakeMessage := messages[:1][0]
+
+		// Per the specifications, if loadBalanced=true, drivers MUST use the hello
+		// command for the initial handshake and use the OP_MSG protocol.
+		assert.Equal(mt, "hello", handshakeMessage.CommandName)
```

Comment:
@blink1073 @qingyang-hu The [handshake specifications ](https://github.com/mongodb/specifications/blob/master/source/mongodb-handshake/handshake.rst#connection-handshake)say that "If a [server API version](https://github.com/mongodb/specifications/blob/master/source/versioned-api/versioned-api.rst) is requested or loadBalanced: True, drivers MUST use the hello command for the initial handshake and use the OP_MSG protocol." So we should always assert that "hello" is used as the command name in this case. 

Changing this lead to the discovery that we were also setting the command name incorrectly on the initial handshake. I've updated that logic in hello.go and relevant SDAM tests per DRIVERS-1929 .